pagopa/io-app
# fix: [IOBP-1101] Add missing `Mixpanel` events/properties

## Short description
This PR focus on enhancing the tracking of payment methods, receipt details, and user interactions adding missing events/properties to `Mixpanel`.

## List of changes proposed in this pull request
- Added `trackPaymentWalletMethodDetail` function to track details of payment methods
- Updated `handleOnMethodPress` function to include tracking of payment method selection and status
- Added `organization_fiscal_code` and `payment_status` to the `trackPaymentsOpenReceipt` function 
- `trackPaymentsSaveAndShareReceipt` to include additional receipt details 
- Added `trackReceiptFilterUsage` function to track the usage of receipt category filters 
- Added tracking for receipt filter usage in `ReceiptListScreen` 
- Updated `ReceiptDetailsScreen` and `ReceiptPreviewScreen` to include more comprehensive analytics tracking when downloading and sharing receipts

## How to test
Verify with `Proxyman` that all data is being transmitted as specified in the open task description
